src/de.rs

```diff
@@ -52,25 +53,71 @@ where
     T::deserialize(Deserializer::new(&value))
 }
 
-impl<'de> IntoDeserializer<'de, Error> for &'de Key {
-    type Deserializer = Deserializer<'de>;
+/// Deserialize the given type from a [Key], with a non-default [`FloatPolicy`](key::FloatPolicy).
+///
+/// # Examples
+///
+/// ```rust
+/// use serde_derive::{Deserialize, Serialize};
+/// use serde_hashkey::{from_key_with_policy, to_key_with_policy, OrderedFloat, Key};
+/// use std::collections::HashMap;
+///
+/// #[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+/// struct Author {
+///     name: String,
+///     age: u32,
+/// }
+///
+/// #[derive(Debug, PartialEq, Eq, Serialize, Deserialize)]
+/// struct Book {
+///     title: String,
+///     author: Author,
+/// }
+///
+/// # fn main() -> serde_hashkey::Result<()> {
+/// let book = Book {
+///     title: String::from("Birds of a feather"),
+///     author: Author {
+///         name: String::from("Noah"),
+///         age: 42,
+///     },
+/// };
+///
+/// let key = to_key_with_policy::<OrderedFloat, _>(&book)?;
+/// let book2 = from_key_with_policy::<OrderedFloat, _>(&key)?;
+///
+/// assert_eq!(book, book2);
+/// # Ok(())
+/// # }
+/// ```
+pub fn from_key_with_policy<Float: FloatPolicy, T>(
```

Comment:
Just a nit, but style-wise I prefer using `where` clauses for trait bounds even though they are semantically the same. But at the very least we should avoid mixing as is done here with `T`.

I also don't believe a separate `from_key_with_policy` is necessary, since the policy is already associated with the `Key` being deserialized 🤔.

Reply:
The reason I added `from_key_with_policy` is that adding the extra type parameter to `from_key` caused the tests to break; I figured I'd err on the side of caution and add a new method w/ no breakage instead. If you'd prefer I can consolidate the `from_key/to_key[_with_policy]` methods into two.